instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/jdbc/DataSourcePostProcessor.java

```diff
@@ -28,7 +33,13 @@ public Object postProcessAfterInitialization(Object bean, String beanName) {
     // Exclude scoped proxy beans to avoid double wrapping
     if (bean instanceof DataSource && !ScopedProxyUtils.isScopedTarget(beanName)) {
       DataSource dataSource = (DataSource) bean;
-      return JdbcTelemetry.create(openTelemetryProvider.getObject()).wrap(dataSource);
+      return JdbcTelemetry.builder(openTelemetryProvider.getObject())
+          .setStatementSanitizationEnabled(
+              configPropertiesProvider
+                  .getObject()
+                  .getBoolean("otel.instrumentation.common.db-statement-sanitizer.enabled", true))
```

Comment:
FYI agent has a specialized flag for jdbc `otel.instrumentation.jdbc.statement-sanitizer.enabled` that defaults to the value of the common sanitizer flag. If you wish to have parity with agent flags you should implement that here also.

Reply:
same logic for mongo?

Reply:
mongo also has `otel.instrumentation.mongo.statement-sanitizer.enabled`

Reply:
is it also using the fallback?

Reply:
it is - found it